numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -314,26 +314,6 @@ npy_PyFile_Check(PyObject *file)
     return 1;
 }
 
-#else
-
-static NPY_INLINE FILE *
```

Comment:
The Python 2 code path is broken because ``PyFile_AsFile(file)`` doesn't work when ``file`` is an ``IOFile`` (as produced by ``io.open``). So the patch removes this code path and simply uses the old Python 3 code path also on Python 2, because that's exactly what the Python 3 code path does: creating ``FILE *`` objects from things like ``IOFile`` and so on.